packages/elements/src/react/sign-up/loading.tsx

```diff
@@ -0,0 +1,116 @@
+import type { OAuthProvider, SamlStrategy } from '@clerk/types';
+import type * as React from 'react';
+
+import { ClerkElementsRuntimeError } from '~/internals/errors';
+import { mapScopeToStrategy } from '~/react/utils/map-scope-to-strategy';
+
+import { useLoading } from '../hooks/use-loading.hook';
+import { SignUpRouterCtx } from './context';
+import { SignUpContinueCtx } from './continue';
+import { SignUpStartCtx } from './start';
+import type { SignUpStep } from './step';
+import { SignUpVerificationCtx } from './verifications';
+
+type Strategy = OAuthProvider | SamlStrategy | 'metamask';
+type LoadingScope = 'global' | `step:${SignUpStep}` | `provider:${Strategy}`;
+
+type LoadingProps = {
+  scope?: LoadingScope;
+  children: (isLoading: boolean) => React.ReactNode;
+};
+
+/**
+ * Access the loading state of a chosen scope. Scope can refer to a step, a provider, or the global loading state. The global loading state is `true` when any of the other scopes are loading.
+ *
+ * @param scope - Optional. Specify which loading state to access. Can be a step, a provider, or the global loading state. If `<Loading>` is used outside a `<Step>` it will default to `"global"`. If used inside a `<Step>` it will default to the current step.
+ * @param {Function} children - A render prop function that receives `isLoading` as an argument. `isLoading` is a boolean that indicates if the current scope is loading or not.
+ *
+ * @example
+ * <SignUp>
+ *   <Loading>
+ *     {(isLoading) => isLoading && "Global loading..."}
+ *   </Loading>
+ * </SignUp>
+ *
+ * @example
+ * <Step name="start">
+ *  <Submit>
+ *    <Loading>
+ *      {(isLoading) => isLoading ? "Start is loading..." : "Submit"}
+ *    </Loading>
+ *  </Submit>
+ * </Step>
+ *
+ * @example
+ * <Step name="start">
+ *   <Loading scope="provider:google">
+ *     {(isLoading) => (
+ *       <Provider name="google" disabled={isLoading}>
+ *         {isLoading ? "Loading..." : "Continue with Google"}
+ *       </Provider>
+ *     )}
+ *   </Loading>
+ * </Step>
+ */
+export function Loading({ children, scope }: LoadingProps) {
```

Comment:
💭 I wonder if we could expose `Loading` from `common` and use a common loading context rendered within each flow. No action needed for now, just a thought! It would likely complicate some of the scope logic by trying to make it generic.